src/main/common/maths.c

```diff
@@ -49,14 +49,33 @@
 #define sinPolyCoef7 -1.980661520e-4f                                          // Double: -1.980661520135080504411629636078917643846e-4
 #define sinPolyCoef9  2.600054768e-6f                                          // Double:  2.600054767890361277123254766503271638682e-6
 #endif
+// float sin_approx(float x)
+// {
+//     int32_t xint = x;
+//     if (xint < -32 || xint > 32) return 0.0f;                               // Stop here on error input (5 * 360 Deg)
+//     while (x >  M_PIf) x -= (2.0f * M_PIf);                                 // always wrap input angle to -PI..PI
+//     while (x < -M_PIf) x += (2.0f * M_PIf);
+//     if (x >  (0.5f * M_PIf)) x =  (0.5f * M_PIf) - (x - (0.5f * M_PIf));   // We just pick -90..+90 Degree
+//     else if (x < -(0.5f * M_PIf)) x = -(0.5f * M_PIf) - ((0.5f * M_PIf) + x);
+//     float x2 = x * x;
+//     return x + x * x2 * (sinPolyCoef3 + x2 * (sinPolyCoef5 + x2 * (sinPolyCoef7 + x2 * sinPolyCoef9)));
+// }
+
+// alternate method suggested by chatgpt
 float sin_approx(float x)
 {
-    int32_t xint = x;
-    if (xint < -32 || xint > 32) return 0.0f;                               // Stop here on error input (5 * 360 Deg)
-    while (x >  M_PIf) x -= (2.0f * M_PIf);                                 // always wrap input angle to -PI..PI
-    while (x < -M_PIf) x += (2.0f * M_PIf);
-    if (x >  (0.5f * M_PIf)) x =  (0.5f * M_PIf) - (x - (0.5f * M_PIf));   // We just pick -90..+90 Degree
-    else if (x < -(0.5f * M_PIf)) x = -(0.5f * M_PIf) - ((0.5f * M_PIf) + x);
+    // Normalize input to the range -PI to PI
```

Comment:
whats wrong with the old sin_approx?

Reply:
Old code was implemented for MCUs without FPU and it was expecting argument close to -2pi..2pi range.

This change is easier to read and should work fine.

Reply:
Thanks, @ledvinap

Reply:
Just remove it. The old code is always still visible on Github. You can always go back and see the code history.